dotnet/src/dotnetframework/GxClasses/Helpers/GXRestUtils.cs

```diff
@@ -14,29 +22,69 @@ internal class GXFormData
 		public static string FORMDATA_REFERENCE = "gxformdataref:";
 	}
 	[System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1812:AvoidUninstantiatedInternalClasses")]
-	internal class GxRestUtil
+	internal class GxUploadHelper
 	{
+
+		internal static bool IsUploadURL(HttpContext httpContext)
+		{
+			return httpContext.Request.GetRawUrl().EndsWith(HttpHelper.GXOBJECT, StringComparison.OrdinalIgnoreCase);
+		}
+		internal static void CacheUploadFile(string fileGuid, string savedFileName, string localfileName, string ext, GxFile file, HttpContext localHttpContext)
+		{
+			CacheAPI.FilesCache.Set(fileGuid, JSONHelper.Serialize(new UploadCachedFile() { path = savedFileName, fileExtension = ext, fileName = localfileName }), GxRestPrefix.UPLOAD_TIMEOUT);
```

Comment:
Just a minor comment. 

Maybe we should consider increase the GxRestPrefix.UPLOAD_TIMEOUT (10 seconds only).

Suppose this scenario: WebPanel with Upload UserControl. 

1. The User Selects and automatically uploades a batch of files
2. The user takes time to complete the Form (many editable fields, validations, etc)
3. After 30 seconds, the User confirms the WebPanel.

In this case the Save will fail, right? Because the Cache Tokens have expired and the Application will not find the Real Path. 
This is not a new bug, but seems to be happening. 

Maybe cache expiration is not enabled (for local cache?), so this don't happen right now?.
However, in Memcached or Redis, I think that cache keys will expire. 

The same case could happen for SD. 

Another approach would have been to use JWE (JWT + Encryption) instead of Cache for storing the Tokens, but maybe too much work to do right now..  But this way we would not need the Cache and not deal with expiration.

Reply:
GxRestPrefix.UPLOAD_TIMEOUT is 10 minutes, it is in minutes. A better timeout would be the same timeout as the http session timeout, wouldn't be? 
Checking if JWE approach was considered before.

Reply:
Oops, sorry! 10 minutes seems more than enough.
Never mind!